### ISSUE TITLE
`matchers/match-with` misbehaves for CLJS numbers

### ISSUE THREAD

**Issue by bryanluu** [nubank/matcher-combinators#202]
When playing around with the `matchers/match-with` method in a CLJS test to set numeric matchers to use `matchers/within-delta`, I discovered an interesting bug. When using `matchers/within-delta` directly, the test behaves as expected. However, when using `matchers/match-with [number? (matchers/within-delta delta)] expected`, a clojurescript test would fail with the message: 

> Error: No protocol method Matcher.-matcher-for defined for type number ,,,

The following test illustrates this issue.
```clojure
(ns example-test
  (:require [matcher-combinators.test] ; adds support for `match?` and `thrown-match?` in `is` expressions
            [matcher-combinators.matchers :as m]
            #?(:clj [clojure.test :refer [deftest testing is]]
               :cljs [cljs.test :refer-macros [deftest testing is]])))

(deftest failing-matcher-test
  (testing "Example numeric test-case"
    (testing "which passes"
      (is (match? (m/within-delta 0.05 1) 0.99)))
    (testing "which errors"
      (is (match? (m/match-with [number? (m/within-delta 0.05)] 1) 0.99)))))
```

I am using `nubank/matcher-combinators {:mvn/version "3.4.0"}`. This behaviour seems to only affect the `number?` checks.

**Comment by philomates** [comment]
Hey, thanks for the issue + reproduction steps.
I found the main issue and am patching it in https://github.com/nubank/matcher-combinators/pull/203 (still a work in progress)

**Comment by philomates** [comment]
I released a fix in 3.8.6, sorry for the delay
